cpp/src/copying/copy.cu

```diff
@@ -183,55 +159,89 @@ class logical_not {
 /**
  * @brief Implementation of copy_if_else() with gather()/scatter().
  *
- * Currently supports only nested-type column_views. Scalars are not supported.
+ * Handles nested-typed columns and scalars
  */
-template <typename Left, typename Right, typename Filter>
-std::unique_ptr<column> scatter_gather_based_if_else(Left const& lhs,
-                                                     Right const& rhs,
+template <typename Filter>
+std::unique_ptr<column> scatter_gather_based_if_else(cudf::column_view const& lhs,
```

Comment:
I think a doxygen block would be a good idea for this function.  It's not clear what "is_left" means at a glance.